addons/sale/models/payment_transaction.py

```diff
@@ -92,11 +92,11 @@ def _log_message_on_linked_documents(self, message):
 
     def _reconcile_after_done(self):
         """ Override of payment to automatically confirm quotations and generate invoices. """
-        sales_orders = self.mapped('sale_order_ids').filtered(lambda so: so.state in ('draft', 'sent'))
         for tx in self:
             tx._check_amount_and_confirm_order()
+        confirmed_sales_orders = self.mapped('sale_order_ids').filtered(lambda so: so.state in ('sale', 'done'))
```

Comment:
```suggestion
        draft_orders = self.sale_order_ids.filtered(lambda so: so.state in ('draft', 'sent'))
        for tx in self:
            tx._check_amount_and_confirm_order()
        confirmed_sales_orders = draft_orders.filtered(lambda so: so.state in ('sale', 'done'))
```
If we pay twice the same order, we don't want to send the confirmation mail twice.